File: 2. python/1. basics/lecture 2/p14.py
class Solution:
    def __init__(self, n): #constructor
        for i in range(1, n + 1):
            for j in range(i):
                print(chr(ord('A') + j), end=' ')
            print()

''' pattern output:

A
A B
A B C
A B C D

'''

if __name__ == "__main__":
    size = int(input("enter size: "))

    print("1. output:")
    Solution(size)

File: 1. cpp/1. basics/lecture 2/p14.cpp

#include <iostream>
using namespace std;
class solution {
public:
    solution(int n) {
      for (int i = 1; i <= n; i++) {
        for (char ch = 'A'; ch < 'A' + i; ch++) {
          cout << ch << " ";
        }
        cout << endl;
      }
    }
};

/* pattern output:

A
A B
A B C
A B C D


*/

int main() {
  int size;
  cout<<"Enter size of pattern:"<<endl;
  cin>>size;

  cout<<"13. :"<<endl;
  solution pattern(size);

}